src/backend/sev/snp/firmware/mod.rs

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Operations for managing the SEV platform.
+
+#[cfg(target_os = "linux")]
+mod linux;
+
+#[cfg(target_os = "linux")]
+pub use linux::Firmware;
+
+use super::Version;
+
+use bitflags::bitflags;
+
+use std::fmt::Debug;
+
+/// The platform state.
+///
+/// The underlying SEV platform behaves like a state machine and can
+/// only perform certain actions while it is in certain states.
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+#[repr(u8)]
+pub enum State {
+    /// The platform is uninitialized.
+    Uninitialized,
+
+    /// The platform is initialized, but not currently managing any
+    /// guests.
+    Initialized,
+}
+
+impl std::fmt::Display for State {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = match self {
+            State::Uninitialized => "uninitialized",
+            State::Initialized => "initialized",
+        };
+        write!(f, "{}", state)
+    }
+}
+
+bitflags! {
+    /// Describes the platform state.
+    #[derive(Default)]
+    pub struct Flags: u32 {
+        /// If set, this platform is owned. Otherwise, it is self-owned.
+        const OWNED           = 1 << 0;
+
+        /// If set, encrypted state functionality is present.
+        const ENCRYPTED_STATE = 1 << 8;
+    }
+}
+
+/// The CPU-unique identifier for the platform.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Identifier(Vec<u8>);
+
+impl From<Identifier> for Vec<u8> {
+    fn from(id: Identifier) -> Vec<u8> {
+        id.0
+    }
+}
+
+impl std::fmt::Display for Identifier {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        for b in self.0.iter() {
+            write!(f, "{:02X}", b)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// Information regarding the SEV-SNP platform's TCB version.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TcbStatus {
+    /// Installed TCB version.
+    pub platform_version: TcbVersion,
+
+    /// Reported TCB version.
+    pub reported_version: TcbVersion,
+}
+
+/// Information regarding the SEV-SNP platform's current status.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Status {
+    /// The build number.
```

Comment:
Is this comment still valid given the type of the field?

Reply:
`build` is of type `Build` defined here https://github.com/enarx/enarx/pull/1026/files#diff-7b889b31bec995c5f7126623e4b70f668d334d68188035dc1274a807a45ff949R107-R116, but comment states it's a *number*

Reply:
changed to 
```rust
    /// The build information.
```